service/settings.py

```diff
@@ -198,13 +209,33 @@ def autodetect(self):
 
     def getProxySettings(self):
 
-        if self.getMode() == 0:
+        if self.getMode() == self.PROXY_MODE_NONE:
             return None
-        if self.getMode() == 1:
+        if self.getMode() == self.PROXY_MODE_AUTODETECT:
             return self.autodetect()
-        if self.getMode() == 2:
+        if self.getMode() == self.PROXY_MODE_MANUAL:
             return (self.getAddress(), int(self.getPort()))
 
+    def getProxyAuthDetails(self):
+        if self.getMode() == self.PROXY_MODE_NONE:
+            return None
+        if (self.serviceNetworkSettings["login"] is None) or (self.serviceNetworkSettings["password"] is None):
+            return None
+        # in all other cases, return tuple of (login, password)
+        return (self.serviceNetworkSettings["login"], self.serviceNetworkSettings["password"])
```

Comment:
`self.serviceNetworkSettings` is already a dict so that should work if you wanted to just return that.

As long as you've commented the code it's not a major issue (like I said, it's a minor quibble).  I prefer the dict as it's easier to follow when tracing the code, but it's up to you.

>I saw similar code

That would be from someone named `HomeWorld` back in 2011.

A _lot_ of Pyfa is *very* old, and some of the services are some of the oldest code around.  Most of `settings.py` is from the 2010-2012 days, and I wouldn't exactly adhere myself to those designs.

Not that it's necessarily bad code (it works fine, and like I said earlier, this is a minor quibble), but a lot has changed in the last 6 years.